samples/nrf9160/multicell_location/src/main.c

```diff
@@ -98,11 +100,24 @@ static void lte_handler(const struct lte_lc_evt *const evt)
 	case LTE_LC_EVT_NEIGHBOR_CELL_MEAS:
 		LOG_INF("Neighbor cell measurements received");
 
-		/* Copy current and neighbor cell information. */
+		if (evt->cells_info.current_cell.id == 0) {
+			LOG_DBG("Cell ID not valid..");
+			break;
+		}
+
+		/* Copy current cell information. */
 		memcpy(&cell_data, &evt->cells_info, sizeof(struct lte_lc_cells_info));
-		memcpy(neighbor_cells, evt->cells_info.neighbor_cells,
-			sizeof(struct lte_lc_ncell) * cell_data.ncells_count);
-		cell_data.neighbor_cells = neighbor_cells;
+
+		/* Copy neighbor cell information. */
+		if (evt->cells_info.ncells_count > 0 && evt->cells_info.neighbor_cells) {
+			memcpy(neighbor_cells,
+			       evt->cells_info.neighbor_cells,
+			       sizeof(struct lte_lc_ncell) * evt->cells_info.ncells_count);
+
+			cell_data.neighbor_cells = neighbor_cells;
+		}
```

Comment:
if `ncells_count == 0`, you could set `cell_data.neighbor_cells = NULL`.  As it is, you leave it pointing at `evt->cells_info.neighbor_cells`, which may or may not be `NULL`.

Reply:
Well, it's good to make sure ncells_count is 0, but what about also making sure neighbor_cells is NULL?

Reply:
cell_data.neighbor_cells is set at init and never modified

Reply:
OK, you also changed from `sizeof(struct lte_lc_cells_info)` to `sizeof(struct lte_lc_cell)` and are only copying the `cells_info.current_cell`.  Previously it copied the whole structure, including the `neighbor_cells` pointer.